src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+/**
+ * Copyright (C) 2022 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "SubCmdConfigure.h"
+#include "tools/common/XBUtilities.h"
+#include "common/system.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+
+constexpr const char* config_file  = "/etc/msd.conf";
+
+enum class config_type {
+    security = 0,
+    clk_scaling,
+    threshold_power_override,
+    threshold_temp_override,
+    reset
+};
+
+enum class mem_type {
+    unknown= 0,
+    ddr,
+    hbm
+};
+
+// Presumably a struct in anticipation of more data
+struct config
+{
+  std::string host;
+
+  void write(std::ostream& ostr)
+  {
+    ostr << "host=" << host << std::endl;
+  }
+};
+
+SubCmdConfigure::SubCmdConfigure(bool _isHidden, bool _isDepricated, bool _isPreliminary)
+    : SubCmd("configure",
+             "Advanced options for configuring a device")
+{
+  const std::string long_description = "Advanced options for configuring a device";
+  setLongDescription(long_description);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+// So far, we only support the following configs, eg.
+// [Device]
```

Comment:
It is evident that this comment is out-of-date with the code.

It would be better to add a mini-comment for each of CONFIG entries prior to checking for its existence.

Reply:
Agreed. There are some options that are supported here and in the hidden options. We can fix the issue now but PRT might need to be involved. It may be good to finish this PR and make another CR to remove the hidden options

Reply:
I removed the older comment, not sure what the mini comments should be